nautobot/extras/views.py

```diff
@@ -1024,6 +1028,7 @@ def post(self, request, class_path=None, slug=None):
                     one_off=schedule_type == JobExecutionType.TYPE_FUTURE,
                     user=request.user,
                     approval_required=job_model.approval_required,
+                    crontab=crontab,
```

Comment:
Now that I'm looking at this again, it appears that there may be a couple ways to do custom-Crontab based schedules incorrectly. Because what if you set `interval=JobExecutionType.TYPE_CUSTOM, one_off=True, but then set `crontab=crontab`. It doesn't appear that we have any checks to assert that these combinations of options are mutualy exclusive.

I would like to see some more defensiveness inside of the `ScheduledJob.clean()` method here and also have these possible combinations tested to make sure that we are thinking about the edge cases and protecting against them with proper error-handling and sanity checks!

Reply:
Hmm, one both `extras/views.py` and `extras/api/views.py`, `one_off` doesn't seem to be set by the user. In either case, it's only ever set to true when the schedule type is equal to `JobExecutionType.TYPE_FUTURE`
I'll see about another defense in the `clean()` method after resolving everything else and testing.